### ISSUE TITLE
Feature Request: Support for server response messages

### ISSUE THREAD

**Issue by mdeknowis** [steveukx/git-js#485]
Sometimes in case for example of git push the server responses with a message, that might be very helpful for the developer.

E.g.:

```txt
remote: 
remote: To create a merge request for MYTEST_BRANCH, visit:
remote:   https://gitlab/group/project/-/merge_requests/new?merge_request%5Bsource_branch%5D=MYTEST_BRANCH
remote:
```

or other kind of information like, why the push was rejected, especially in case of server side hooks

So it would very nice to be able to consume these information, so own code can either process it or at least print it out.
So perhaps something similar to <https://nodejs.org/api/child_process.html#child_process_child_process_execsync_command_options> , you get a string or buffer back etc.
Or perhaps any other chance to get that output like using callbacks or streams.

**Comment by steveukx** [comment]
Hi, thank you for the suggestion.

[pull](https://github.com/steveukx/git-js/blob/v2.13.2/typings/simple-git.d.ts#L383) now returns a [PushResult](https://github.com/steveukx/git-js/blob/v2.13.2/typings/response.d.ts#L364-L378) that has a [remoteMessages](https://github.com/steveukx/git-js/blob/v2.13.2/typings/response.d.ts#L343-L351) property with some details parsed out for you (`pullRequestUrl` detected in github / gitlab format, and vulnerabilities detected in github format, if gitlab does this as well, please do open an issue/pull request with the syntax they use) and an `all` array of each `remote: ` line in the response from the git remote.

If you do need to get to the underlying data from a `simple-git` instance, you can use the `.outputHandler` function to pipe the `stdout` / `stderr` streams to some other process [example](https://github.com/steveukx/git-js/tree/v2.13.2#piping-to-the-console-for-long-running-tasks)

**Comment by mdeknowis** [comment]
@steveukx  Any chance to get this for push, too?

**Comment by steveukx** [comment]
I had a typo in my previous reply... it was applied to `push` in `2.13.2`, and now also available for `push` and `merge` as of `2.17.0`.
If you have any problems, or have examples of the `remote: ` messages that would be useful to add specific parsers for, please open another issue.
